### ISSUE TITLE
CoordConv

### ISSUE THREAD

**Issue by rytisss** [tensorflow/tensorflow#42621]
**System information**
- TensorFlow version 2.3.0 and everything latest and greatest:
- Are you willing to contribute it (Yes/No): Depends, there are already a few written by developers that might be a solution for this request.

**Describe the feature and the current behavior/state.**
Basically a gradient matrices before the convolutional operation (Conv1D, Conv2D, Conv3D) that gives more information about convolutional operation place in the image.

**Will this change the current api? How?**
I don't think so. CoordConv can be added as a separate layer before the convolutional operation.

**Who will benefit with this feature?**
In general, all the segmentation or detection task in which objects transition in the image is important

**Any Other info.**
Article: [https://arxiv.org/abs/1807.03247](https://arxiv.org/abs/1807.03247) and original implementation by authors [https://github.com/uber-research/CoordConv](https://github.com/uber-research/CoordConv)
Video explanation: [https://www.youtube.com/watch?v=8yFQc6elePA](https://www.youtube.com/watch?v=8yFQc6elePA)
Possible implementation: [https://github.com/mvoelk/keras_layers](https://github.com/mvoelk/keras_layers)
Similar feature issue/mention : #32222

**Comment by rytisss** [comment]
Hi,
Regarding CoordConv implementation, I would at the service of contributing even in C++ side :)

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
